src/main/java/com/chnu/util/LoggedUserUtil.java

```diff
@@ -0,0 +1,19 @@
+package com.chnu.util;
+
+import com.chnu.exception.UserNotFoundException;
+import com.sun.security.auth.UserPrincipal;
+import org.springframework.security.core.context.SecurityContextHolder;
+import java.util.Optional;
+
+public class LoggedUserUtil {
+
+    public static Optional<UserPrincipal> getLoggedUser() {
+        Object principal = SecurityContextHolder.getContext().getAuthentication().getPrincipal();
+        if (principal instanceof UserPrincipal) {
+            return Optional.of((UserPrincipal) principal);
+        }
```

Comment:
Are you assured that we must use Optional<> in this plase?

Reply:
Your English is very well :)

Reply:
But question is good